Pull Request Messenger

RobinE89 opened the pull request.

wouterfolkertsma committed:
Fix

Commit:
gitflow-feature-stash: CustomAlerts

Commit:
Merge branch 'feature/CustomAlerts' into develop

Commit:
Merge remote-tracking branch 'origin/develop' into develop

Commit:
Upload Drag&Drop

Commit:
Upload-Update + Manage + Delete

Commit:
ChartJS

Commit:
Merge branch 'feature/upload' into develop

Commit:
Fix DataUploadRefresh

wouterfolkertsma committed:
Cleanups

Commit:
gitflow-feature-stash: messenger

wouterfolkertsma commented on app/Http/Controllers/MessengerController.php:
```diff
@@ -0,0 +1,82 @@
+        //EmployeeVersion (Can see All!)
+        //$users = User::all();
+
+        $users = DB::select("select users.id, users.first_name, users.last_name, users.email, count(is_read) as unread
```
This can be done by the querybuilder.

wouterfolkertsma commented on resources/views/messenger/inbox.blade.php:
```diff
@@ -0,0 +1,155 @@
+
+                        console.log("Message - > Send")
+                        $('#' + data.from).click();
+                        scrollToBottom();
+                    },
+                    error: function (jqXHR, status, err){
+
+                    },
+                    complete:function (){
+
+                    }
+                })
+            }
+        });
+    });
+
+    function scrollToBottom(){
+        $('.message-wrapper').animate({
+            scrollTop: $('.message-wrapper').get(0).scrollHeight
+        }, 50);
+    }
+
+    function newMessageRecieved(){
+        const Toast = Swal.mixin({
+            toast: true,
+            position: 'top-end',
+            showConfirmButton: false,
+            timer: 3000,
+            timerProgressBar: true,
+            didOpen: (toast) => {
+                toast.addEventListener('mouseenter', Swal.stopTimer)
+                toast.addEventListener('mouseleave', Swal.resumeTimer)
+            }
+        })
+        Toast.fire({
+            icon: 'info',
+            title: "received new message!"
+        })
+    }
+</script>
```
This should be a javascript component

wouterfolkertsma commented on resources/sass/components/messanger.scss:
```diff
@@ -0,0 +1,111 @@
+.messenger-inbox {
```
Messanger = messenger

wouterfolkertsma commented on database/migrations/2020_12_10_103112_create_messages_table.php:
```diff
@@ -0,0 +1,35 @@
+    public function up()
+    {
+        Schema::create('messages', function (Blueprint $table) {
+            $table->id();
+            $table->bigInteger( 'from');
+            $table->bigInteger( 'to');
+            $table->text( 'message');
+            $table->tinyInteger('is_read');
+            $table->timestamps();
```
I am missing the foreign key constraints here.. There is no database relationship between the user and messages

wouterfolkertsma commented on database/factories/MessageFactory.php:
```diff
@@ -0,0 +1,39 @@
+    public function definition()
+    {
+        return[
+            'from' => rand(1,17),
+            'to' => rand(1,17),
```
Not the way to go here, no real relation with the user here.

wouterfolkertsma committed:
Edited the messages database layout

Commit:
update (error)

Commit:
update

wouterfolkertsma committed:
Yooooo fixed ur code

Commit:
...fixed ur code bra.....

Commit:
Finish for now

huge security issue @wouter (you changed that)-> if you add the message to the broadcast-message-> then of course everyone can read them. the trigger is only used to update the UI and get the message from the database.
